src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothGatt.cs

```diff
@@ -257,23 +257,164 @@ protected virtual void Dispose(bool disposing)
+        private TaskCompletionSource<bool> _taskForDisconnection;
+        private static event EventHandler<GattConnectionStateChangedEventArgs> s_connectionStateChanged;
+        private static Interop.Bluetooth.GattConnectionStateChangedCallBack s_connectionStateChangeCallback;
 
         internal BluetoothGattClient(string remoteAddress)
         {
             _impl = new BluetoothGattClientImpl(remoteAddress);
             _remoteAddress = remoteAddress;
+            StaticConnectionStateChanged += OnConnectionStateChanged;
         }
 
-        internal static BluetoothGattClient CreateClient(string remoteAddress)
+        /// <summary>
+        /// Creates the Bluetooth GATT client.
+        /// </summary>
+        /// <returns>The BluetoothGattClient instance.</returns>
+        /// <feature>http://tizen.org/feature/network.bluetooth.le.gatt.client</feature>
+        /// <exception cref="NotSupportedException">Thrown when the BT/BTLE is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the create GATT client fails.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public static BluetoothGattClient CreateClient(string remoteAddress)
         {
             BluetoothGattClient client = new BluetoothGattClient(remoteAddress);
             return client.Isvalid() ? client : null;
         }
 
+        /// <summary>
+        /// The ConnectionStateChanged event is raised when the gatt connection state is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<GattConnectionStateChangedEventArgs> ConnectionStateChanged;
+
+        private void OnConnectionStateChanged(Object s, GattConnectionStateChangedEventArgs e)
+        {
+            if (e.RemoteAddress == _remoteAddress)
+            {
+                if (_taskForConnection != null && !_taskForConnection.Task.IsCompleted)
+                {
+                    if (e.Result == (int)BluetoothError.None)
+                    {
+                        _taskForConnection.SetResult(true);
+                    }
+                    else
+                    {
+                        _taskForConnection.SetException(BluetoothErrorFactory.CreateBluetoothException((int)e.Result));
+                    }
+                    _taskForConnection = null;
+                }
+
+                if (_taskForDisconnection != null && !_taskForDisconnection.Task.IsCompleted)
+                {
+                    if (e.Result == (int)BluetoothError.None)
+                    {
+                        _taskForDisconnection.SetResult(true);
+                    }
+                    else
+                    {
+                        _taskForDisconnection.SetException(BluetoothErrorFactory.CreateBluetoothException(e.Result));
+                    }
+                    _taskForDisconnection = null;
+                }
+
+                ConnectionStateChanged?.Invoke(this, e);
```

Comment:
How about only raise Connection StateChanged event when Result is success?
Because, when Result is failed, State will not changed

Reply:
Thank you.